ChiModules/LinearBoltzmannSolver/lbs_linear_boltzmann_solver.h

```diff
@@ -186,6 +190,17 @@ class Solver : public chi_physics::Solver
   void ZeroOutflowBalanceVars(LBSGroupset& groupset);
   void ComputeBalance();
 
+  //
+  void ZeroAngularFluxDataStructures(LBSGroupset& groupset)
+  {
+    if (options.save_angular_flux)
+    {
+      auto& psi = psi_new_local[groupset.id];
+      psi.assign(psi.size(), 0.0);
+    }
+    groupset.angle_agg.ZeroOutgoingDelayedPsi();
+  }
```

Comment:
Every segment of code that involves a call to `ZeroAngularFluxDataStructures` is followed immediately by an assignment of zero to `phi_new_local`. Can we not combine these operations into a common `ZeroFluxDataStrucures` for questions of modularity and readability? The new method might read as:
```
void ZeroFluxDataStructures(LBSGroupset& groupset)
{
  phi_new_local.assign(phi_new_local.size(), 0.0);
  if (options.save_angular_flux)
    psi_new_local[groupset.id].assign(psi_new_local[groupset.id].size(), 0.0);
  groupset.angle_agg.ZeroOutgoingDelayedPsi();
}
```
Also, there is the question as to whether or not it is necessary to zero the angular flux between iterations. In our current approach, I do not believe it to be necessary. However, if we do retain the practice of zeroing-out the downstream angular flux, then it might be useful to zero-out the downstream scalar flux in the same method.

Reply:
I like this idea. A future pull request I will put up after this does more to organize the sweep preparation routines and this will be a nice addition to it.